pkg/libartifact/store/store.go

```diff
@@ -200,26 +203,33 @@ func (as ArtifactStore) Add(ctx context.Context, dest string, paths []string, _
 		if err != nil {
 			return nil, err
 		}
-		newArtifactAnnotations := map[string]string{}
-		newArtifactAnnotations[specV1.AnnotationTitle] = filepath.Base(path)
+		// currently we don't allow override of the filename
+		if _, hasTitle := options.Annotations[specV1.AnnotationTitle]; hasTitle {
```

Comment:
I would check this before copying the blob to the storage. Don't pollute the storage with dangling blobs.

Reply:
yes good idea